compiler/qsc/src/interpret.rs

```diff
@@ -345,17 +349,43 @@ impl Interpreter {
     /// Performs QIR codegen using the given entry expression on a new instance of the environment
     /// and simulator but using the current compilation.
     pub fn qirgen(&mut self, expr: &str) -> std::result::Result<String, Vec<Error>> {
-        if self.capabilities != RuntimeCapabilityFlags::empty() {
+        if self.capabilities == RuntimeCapabilityFlags::all() {
             return Err(vec![Error::UnsupportedRuntimeCapabilities]);
         }
+        if self.capabilities == RuntimeCapabilityFlags::empty() {
+            let mut sim = BaseProfSim::new();
+            let mut stdout = std::io::sink();
+            let mut out = GenericReceiver::new(&mut stdout);
 
-        let mut sim = BaseProfSim::new();
-        let mut stdout = std::io::sink();
-        let mut out = GenericReceiver::new(&mut stdout);
-
-        let val = self.run_with_sim(&mut sim, &mut out, expr)??;
+            let val = self.run_with_sim(&mut sim, &mut out, expr)??;
 
-        Ok(sim.finish(&val))
+            Ok(sim.finish(&val))
+        } else {
+            // Compile the expression. This operation will set the expression as
+            // the entry-point in the FIR store.
+            let (graph, compute_properties) = self.compile_entry_expr(expr)?;
+            // The above line updated the entry expr in the FIR store, but we need to update
+            // the user package exec graph.
+            let package = self.fir_store.get_mut(self.package);
+            package.entry_exec_graph = graph.into();
```

Comment:
This doesn't seem consistent with how we handle entry expressions elsewhere. It's also causing the `fir_store` to get "out of sync" with the HIR package store - the HIR package store doesn't have an entry point the FIR store does. Though this is probably harmless now, it's just one more thing to keep track of mentally that I'm worried might cause bugs as we refactor or add new features down the line.

I think `fir_to_qir` should take the entry exec graph, just like `eval` does. Then it can be plumbed all the way to the partial evaluator.  Easier to reason about IMO.

Later, if we ever want to tweak how entry expressions are handled, we don't have to worry about the evaluator and partial evaluator separately since they'll be explicit and consistent in their handling.